library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -2273,6 +2274,469 @@ void libspdm_set_last_spdm_error_struct(void *spdm_context, libspdm_error_struct
+                     sizeof(libspdm_fips_selftest_context),
+                     &(libspdm_context->fips_selftest_context),
+                     sizeof(libspdm_fips_selftest_context));
+    return true;
+}
+
+/**
+ * update fips_selftest_context in spdm_context;
+ *
+ * @param[in]    spdm_context                A pointer to the spdm_context.
+ *
+ * @retval true   run FIPS self_test successfully;
+ * @retval false  run FIPS self_test failed;
+ */
+bool libspdm_update_fips_selftest_context(void *spdm_context)
+{
+    libspdm_context_t *context;
+    bool result;
+
+    result = true;
+    context = spdm_context;
+    if (context == NULL) {
+        result = false;
+    }
+
+#if LIBSPDM_FIPS_TEST_AT_LOAD
+    if (context->fips_selftest_context.tested_algo == 0) {
+        if (!libspdm_fips_run_selftest()) {
+            result = false;
+        } else {
+            /*all algos are tested and the all results passed*/
+            context->fips_selftest_context.tested_algo |=
+#if LIBSPDM_SHA256_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_HMAC_SHA256 |
+#endif/*LIBSPDM_SHA256_SUPPORT*/
+
+#if LIBSPDM_SHA384_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_HMAC_SHA384 |
+#endif/*LIBSPDM_SHA384_SUPPORT*/
+
+#if LIBSPDM_SHA512_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_HMAC_SHA512 |
+#endif/*LIBSPDM_SHA512_SUPPORT*/
+
+#if LIBSPDM_AEAD_GCM_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_AES_GCM |
+#endif/*LIBSPDM_AEAD_GCM_SUPPORT*/
+
+#if LIBSPDM_RSA_SSA_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_RSA_SSA |
+#endif/*LIBSPDM_RSA_SSA_SUPPORT*/
+
+#if LIBSPDM_SHA256_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_HKDF |
+#endif/*LIBSPDM_SHA256_SUPPORT*/
+
+#if LIBSPDM_ECDHE_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_ECDH |
+#endif/*LIBSPDM_ECDHE_SUPPORT*/
+
+#if LIBSPDM_SHA256_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_SHA256 |
+#endif/*LIBSPDM_SHA256_SUPPORT*/
+
+#if LIBSPDM_SHA384_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_SHA384 |
+#endif/*LIBSPDM_SHA384_SUPPORT*/
+
+#if LIBSPDM_SHA512_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_SHA512 |
+#endif/*LIBSPDM_SHA512_SUPPORT*/
+
+#if (LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_256_SUPPORT)
+                LIBSPDM_FIPS_SELF_TEST_SHA3_256 |
+#endif/*(LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_256_SUPPORT)*/
+
+#if (LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_384_SUPPORT)
+                LIBSPDM_FIPS_SELF_TEST_SHA3_384 |
+#endif/*(LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_384_SUPPORT)*/
+
+#if (LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_512_SUPPORT)
+                LIBSPDM_FIPS_SELF_TEST_SHA3_512 |
+#endif/*(LIBSPDM_SHA3_SUPPORT_TEST && LIBSPDM_SHA3_512_SUPPORT)*/
+
+#if LIBSPDM_FFDHE_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_FFDH |
+#endif/*LIBSPDM_FFDHE_SUPPORT*/
+
+#if LIBSPDM_ECDSA_SUPPORT
+                LIBSPDM_FIPS_SELF_TEST_ECDSA |
+#endif/*LIBSPDM_ECDSA_SUPPORT*/
+
+#if LIBSPDM_EDDSA_SUPPORT_TEST
+                LIBSPDM_FIPS_SELF_TEST_EDDSA
```

Comment:
this line is missing "|", did it compile?

Reply:
Sorry. I forget to add the "|". The code has added. Thanks.